### ISSUE TITLE
ERROR in Solaris SPARC

### ISSUE THREAD

**Issue by Enchufa2** [r-simmer/simmer.plot#2]
What we know:

* It has something to do with DiagrammeR:

```
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
  4: identical(as.vector(object), TRUE)
  5: as.vector(object)
  6: inherits(plot(x), "htmlwidget")
  7: plot(x)
  8: plot.trajectory(x)
  9: DiagrammeR::trav_in(graph)
```

* which means that the error is [here](https://github.com/r-simmer/simmer.plot/blob/02d70ad96aaccd31e569cf786bf5507521a22620/R/plot.trajectory.R#L115).
* More info from the `plot.trajectory()` example, which uses the same trajectory:

```
Warning in is.na(get_predecessors(graph, selected_nodes[i])[1]) :
  is.na() applied to non-(list or vector) of type 'NULL'
Error in if (!is.na(get_predecessors(graph, selected_nodes[i])[1])) { : 
  argument is of length zero
```

One option would be implement our own graph _navigator_ to avoid `DiagrammeR::trav_in()`.

**Comment by Enchufa2** [comment]
This probably is (was) a bug in DiagrammeR: [this is the old code that causes the error](https://github.com/rich-iannone/DiagrammeR/blob/b7d1f13c6170d8229ffd0d4d137aec654e6395e1/R/trav_in.R#L120) (v0.8.4 on CRAN now), but [new code is totally rewritten](https://github.com/rich-iannone/DiagrammeR/blob/072d24a2bee59721a83bf4dfad1012267f0b0100/R/trav_in.R) (v0.9.0 still to appear on CRAN). So let's wait.

**Comment by Enchufa2** [comment]
simmer.plot 0.1.5 still erroring with DiagrammeR 0.9.0. Some improvement though: now, the tests are failing: 

```
 > test_check("simmer.plot")
  1. Failure: a simple trajectory is correctly converted to graph (@test-plot-trajectory.R#62) 
  all(nodes$label == name) isn't true.
  2. Failure: a simple trajectory is correctly converted to graph (@test-plot-trajectory.R#62) 
  1 is not null.
  3. Failure: a simple trajectory is correctly converted to graph (@test-plot-trajectory.R#62) 
  2 is not null.
  4. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#68) 
  all(nodes$label == name) isn't true.
  5. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#68) 
  all(edges$from == from) isn't true.
  6. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#68) 
  all(edges$to == to) isn't true.
  7. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#69) 
  all(nodes$label == name) isn't true.
  8. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#69) 
  all(edges$from == from) isn't true.
  9. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#69) 
  all(edges$to == to) isn't true.
  10. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#70) 
  all(nodes$label == name) isn't true.
  11. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#70) 
  all(edges$from == from) isn't true.
  12. Failure: a rollback with variable amount is correctly converted to graph (@test-plot-trajectory.R#70) 
  all(edges$to == to) isn't true.
  13. Failure: a complex trajectory is correctly converted to graph (@test-plot-trajectory.R#100) 
  all(nodes$label == name) isn't true.
  14. Failure: a complex trajectory is correctly converted to graph (@test-plot-trajectory.R#100) 
  all(edges$from == from) isn't true.
  15. Failure: a complex trajectory is correctly converted to graph (@test-plot-trajectory.R#100) 
  all(edges$to == to) isn't true.
```

**Comment by Enchufa2** [comment]
The problem still persists with v0.1.7, v0.1.8.

**Comment by Enchufa2** [comment]
I got my hands on a SPARC machine, and this is the output from `simmer`:

```r
> library(simmer)
> x <- trajectory() %>%
+     seize("res0", 1) %>%
+     branch(function() 1, c(TRUE, FALSE),
+            trajectory() %>%
+              clone(2,
+                    trajectory() %>%
+                      seize("res1", 1) %>%
+                      timeout(1) %>%
+                      release("res1", 1),
+                    trajectory() %>%
+                      trap("signal",
+                           handler=trajectory() %>%
+                             timeout(1)) %>%
+                      timeout(1)),
+            trajectory() %>%
+              set_attribute("dummy", 1) %>%
+              seize("res2", function() 1) %>%
+              timeout(function() rnorm(1, 20)) %>%
+              release("res2", function() 1) %>%
+              release("res0", 1) %>%
+              rollback(11)) %>%
+     synchronize() %>%
+     rollback(2) %>%
+     release("res0", 1)
> x$verbose=T
> x
trajectory: anonymous, 18 activities
{ Activity: Seize |         0 <- 0x1585cb0 -> 0xf51d40  | resource: res0 | amount: 1 }{ Activity: Branch | 0x1585cb0 <-  0xf51d40 -> 0x84c358  | option: 0x14f7fd0 }  Fork 1, continue,  trajectory: anonymous, 7 activities
  { Activity: Clone |  0xf51d40 <-  0x933ab0 -> 0x84c358  | n: 2 }    Fork 1, continue,    trajectory: anonymous, 3 activities
    { Activity: Seize |  0x933ab0 <-  0xc75d98 -> 0xdf7258  | resource: res1 | amount: 1 }    { Activity: Timeout |  0xc75d98 <-  0xdf7258 -> 0x71e018  | delay: 1 }    { Activity: Release |  0xdf7258 <-  0x71e018 -> 0x84c358  | resource: res1 | amount: 1 }    Fork 2, continue,    trajectory: anonymous, 3 activities
    { Activity: Trap |  0x933ab0 <-  0xd0d688 -> 0xedfc08  | signals: [signal] }      Fork 1, stop,      trajectory: anonymous, 1 activities
      { Activity: Timeout |  0xd0d688 <- 0x1485028 -> 0         | delay: 1 }    { Activity: Timeout |  0xd0d688 <-  0xedfc08 -> 0x84c358  | delay: 1 }  Fork 2, stop,  trajectory: anonymous, 6 activities
  { Activity: SetAttribute |  0xf51d40 <- 0x1022ff0 -> 0xe76418  | key: dummy, value: 1, global: 0 }  { Activity: Seize | 0x1022ff0 <-  0xe76418 -> 0x7c0cc0  | resource: res2 | amount: 0x9cf5e0 }  { Activity: Timeout |  0xe76418 <-  0x7c0cc0 -> 0xe4d208  | delay: 0x6e7788 }  { Activity: Release |  0x7c0cc0 <-  0xe4d208 -> 0x13e3bf0 | resource: res2 | amount: 0xda11d4 }  { Activity: Release |  0xe4d208 <- 0x13e3bf0 -> 0xb2fe20  | resource: res0 | amount: 1 }  { Activity: Rollback | 0x13e3bf0 <-  0xb2fe20 -> 0         | amount: 11 (Seize), times: 1 }{ Activity: Synchronize |  0xf51d40 <-  0x84c358 -> 0x9d14e0  | wait: 1 }{ Activity: Rollback |  0x84c358 <-  0x9d14e0 -> 0x7faf70  | amount: 2 (Branch), times: 1 }{ Activity: Release |  0x9d14e0 <-  0x7faf70 -> 0         | resource: res0 | amount: 1 }
```

Namely, there are no newlines from the C++ core! Bug in `simmer` or `Rcpp`?

**Comment by Enchufa2** [comment]
The ERROR in SPARC was due to RcppCore/Rcpp#647. Solved in RcppCore/Rcpp#649.
